includes/blocks/class-sensei-progress-bar-block.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * File containing the Sensei_Progress_Bar_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Progress_Bar_Block
+ */
+class Sensei_Progress_Bar_Block {
+
+	/**
+	 * Sensei_Progress_Bar_Block constructor.
+	 */
+	public function __construct() {
+		add_action( 'enqueue_block_assets', [ $this, 'enqueue_block_assets' ] );
+		add_action( 'init', [ $this, 'register_block' ] );
+	}
+
+	/**
+	 * Enqueue frontend and admin assets.
+	 *
+	 * @access private
+	 */
+	public function enqueue_block_assets() {
+		if ( 'course' !== get_post_type() ) {
+			return;
+		}
+
+		Sensei()->assets->enqueue( 'sensei-progress-bar', 'blocks/progress-bar/index.js' );
```

Comment:
We won't need the JS in the frontend, this could be in the `enqueue_block_editor_assets` hook to only load it for the editor.

Reply:
Good spot! Added a check to avoid the extra method/hook: e46e5b8891014b76621d53aa95650fc33bb0648c